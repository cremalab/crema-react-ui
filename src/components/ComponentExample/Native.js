// import { withTheme } from 'styled-components'
import Native, { withTheme } from 'styled-components/native'
import ComponentFactory from '.'

export const Component = ComponentFactory({elements: Native, platform: 'native'})
export default withTheme(Component)
