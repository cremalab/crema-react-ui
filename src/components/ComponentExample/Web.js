import { withTheme } from 'styled-components'
import Web from 'styled-components'
import ComponentFactory from '.'

export const Component = ComponentFactory({elements: Web, platform: 'web'})
export default withTheme(Component)
