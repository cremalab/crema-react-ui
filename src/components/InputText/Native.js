import { withTheme } from 'styled-components'
import styled from 'styled-components/native'
import ComponentFactory from '.'
import { Box } from '../indexNative'

export const Component = ComponentFactory({
  elements: {
    styled,
    Box
  },
  platform: 'native'
})
export default withTheme(Component)
