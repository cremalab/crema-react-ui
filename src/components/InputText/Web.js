import styled, { withTheme } from 'styled-components'
import ComponentFactory from '.'
import { Box } from '../indexWeb'

export const Component = ComponentFactory({
  elements: {
    styled,
    Box
  },
  platform: 'web'
})
export default withTheme(Component)
