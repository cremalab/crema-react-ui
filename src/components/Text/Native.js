import { withTheme } from 'styled-components'
import Native from 'styled-components/native'
import TextFactory from '.'

export const Text = TextFactory({elements: Native, platform: 'native'})
export default withTheme(Text)
