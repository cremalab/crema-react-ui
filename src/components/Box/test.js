import React from 'react'
import { Box as BoxWeb } from './Web'
import { Box as BoxNat } from './Native'
import { shallow } from 'enzyme'

describe('<Box /> Web', () => {


  describe('render', () => {
    it('without children', () => {
      const wrapper = shallow(<BoxWeb />)
      expect(wrapper.length).toBe(1)
    })
    it('with children', () => {
      const wrapper = shallow(<BoxWeb>Hi</BoxWeb>)
      expect(wrapper.find('Child').shallow().text())
        .toBe('Hi')
    })
  })

  describe('props', () => {
    it(`default`, () => {
      const wrapper = shallow(
        <BoxWeb>
          Hi
        </BoxWeb>
      )
      expect(wrapper.unrendered.props).toMatchSnapshot()
    })
    it(`backgroundColor`)
    it(`backgroundImage`)
    it(`borderColor`)
    it(`borderWidth`)
    it(`borderRadius`)
    it(`children`)
    it(`margin`)
    it(`opacity`)
    it(`padding`)
    it(`theme`)
    it(`childSpacing`)
    it(`childLayout`)
    it(`childAlign`)
    it(`childWrap`)
    it(`childJustify`)
  })

})

describe('<Box /> Native', () => {

  describe('render', () => {
    it('without children', () => {
      const wrapper = shallow(<BoxNat />)
      expect(wrapper.length).toBe(1)
    })
    // it('with children', () => {
    //   const wrapper = render(<BoxNat><BoxNat/></BoxNat>)
    //   expect(wrapper.length)
    //     .toBe(1)
    // })
  })

  describe('theme prop', () => {
    it(`backgroundColor`)
    it(`backgroundImage`)
    it(`borderColor`)
    it(`borderWidth`)
    it(`borderRadius`)
    it(`children`)
    it(`margin`)
    it(`opacity`)
    it(`padding`)
    it(`theme`)
  })

  describe('convenience prop', () => {
    it(`childSpacing`)
    it(`childLayout`)
    it(`childAlign`)
    it(`childWrap`)
    it(`childJustify`)
  })

})
