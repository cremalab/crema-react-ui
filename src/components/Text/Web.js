import { withTheme } from 'styled-components'
import Web from 'styled-components'
import TextFactory from '.'

export const Text = TextFactory({elements: Web, platform: 'web'})
export default withTheme(Text)
