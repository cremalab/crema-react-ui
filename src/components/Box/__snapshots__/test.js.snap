exports[`<Box /> Web props default 1`] = `
Object {
  "childJustify": "start",
  "childLayout": "column",
  "childSpacing": 0,
  "children": "Hi",
  "min": false,
  "opacity": 1,
  "theme": Object {
    "color": Object {},
    "opacity": Object {},
    "spacing": Object {
      "unit": "px",
    },
  },
}
`;
